compiler/damlc/lib/DA/Cli/Damlc/Packaging.hs

```diff
@@ -458,14 +459,24 @@ buildLfPackageGraph targetLfVersion pkgs stablePkgs dependencyPkgs = (depGraph,
             | dalfPkg <- MS.elems dependencyPkgs <> MS.elems stablePkgs <> map decodedDalfPkg pkgs
             ]
 
+    allPackageRefs :: LF.Package -> [LF.PackageRef]
+    allPackageRefs pkg =
+      toListOf packageRefs pkg
+        <>
+        [ qualPackage
```

Comment:
adding the packages from the `$$imports` value seems to do the trick, though I don't like that this means we call `decodeModuleImports` from two places (here + DataDependencies)

Reply:
I agree with @sofiafaro-da that it's probably better to add a constructor for imports (or perhaps only orphan imports) to the AST, since then we can just leave this unchanged. Also, I'm not sure if other calls to `packageRefs` also need to be amended, but having the orphan imports in the AST would take care of that automatically.

Reply:
I was thinking of the former but it's true that wouldn't work for older LF versions. The latter hadn't occurred to me but it does sound a bit icky. 

It's probably out of the question to go with the former and drop support for data-dependencies older than x, right?

as a third option, perhaps we can use something like Trees That Grow so the main decoder stays to spec, but internally we use a parametrized type that does have a field for orphan imports?

Reply:
> It's probably out of the question to go with the former and drop support for data-dependencies older than x, right?

I’d say so yes. Even in SDK 2.0 we only support 1.14 and given the timeline there is no chance of getting a new LF release out there and once we release it in SDK 2.0 we’re going to support it for a long time.

> as a third option, perhaps we can use something like Trees That Grow so the main decoder stays to spec, but internally we use a parametrized type that does have a field for orphan imports?

That could work although I think I’m not sure it buys us that much assuming the places where we need to care about this is really only the one modified in this PR. I think I’d stick with what you have for now and revisit if it becomes an issue later.

Reply:
My idea would be to have package dependencies specified at the top level of an LF package, rather than dispersed among the AST as we currently do. That would let us add additional package dependencies that don't get used explicitly, like the ones in the imports, (and probably speed up the processing of data-dependencies a little bit, since we don't need to traverse the entire AST to figure out what packages this package depends on).

This wouldn't solve the issue for older LF versions, so the current solution is good to have either way.